### ISSUE TITLE
how to deal with the non-square inputs

### ISSUE THREAD

**Issue by cxzhou35** [hmichaeli/alias_free_convnets#4]
My input image has the shape of [B, C, H, W], etc. [1, 3, 1280, 720]. 
In your codes, it seems like the LPF_RECON_RFFT only using the last dim(W here) to build a 2D frequency mask, how to modify the coeds to fit the non-square inputs? Thanks in advance.

**Comment by hmichaeli** [comment]
Thank you for raising this question.
You are correct - the current implementation in this repository assumes square inputs and will not work on non-square images.
We addressed the more general case (including non-square inputs) in [another project](https://github.com/HadarRosenberg/Shift_Equivariant_Object_Detection/blob/main/project/project_utils/ideal_lpf.py). I hope you find it useful.

**Comment by cxzhou35** [comment]
Thanks for your quick reply!!
My copilot agent actually has done this, it seems like the code is accurate according to the codes you shared.

```python
def create_lpf_rect_hw(H, W, cutoff=0.5):
    cutoff_low_h = int((H * cutoff) // 2)
    cutoff_high_h = int(H - cutoff_low_h)
    rect_h = torch.ones(H)
    rect_h[cutoff_low_h + 1:cutoff_high_h] = 0
    if H % 4 == 0:
        # if N is divides by 4, nyquist freq should be 0
        # N % 4 =0 means the downsampeled signal is even
        rect_h[cutoff_low_h] = 0
        rect_h[cutoff_high_h] = 0

    cutoff_low_w = int((W * cutoff) // 2)
    cutoff_high_w = int(W - cutoff_low_w)
    rect_w = torch.ones(W)
    rect_w[cutoff_low_w + 1:cutoff_high_w] = 0
    if W % 4 == 0:
        # if N is divides by 4, nyquist freq should be 0
        # N % 4 =0 means the downsampeled signal is even
        rect_w[cutoff_low_w] = 0
        rect_w[cutoff_high_w] = 0

    rect_2d = rect_h[:, None] * rect_w[None, :]
    return rect_2d
```
